lib/auth/trustedcluster.go

```diff
@@ -54,17 +54,21 @@ func (a *Server) UpsertTrustedCluster(ctx context.Context, tc types.TrustedClust
 	// It is recommended to omit trusted cluster name because the trusted cluster name
 	// is updated to the roots cluster name during the handshake with the root cluster.
 	var existingCluster types.TrustedCluster
-	var cas []types.CertAuthority
+	var foundTrustedCluster bool
 	if tc.GetName() != "" {
 		var err error
 		existingCluster, err = a.GetTrustedCluster(ctx, tc.GetName())
 		if err != nil && !trace.IsNotFound(err) {
 			return nil, trace.Wrap(err)
 		}
+
+		if err == nil {
+			foundTrustedCluster = true
+		}
```

Comment:
This reads better to me but I can't tell if it's just Go brain rot.

```suggestion
	if tc.GetName() != "" {
		ec, err := a.GetTrustedCluster(ctx, tc.GetName())
		if err == nil {
			existingCluster = ec
		} else if !trace.IsNotFound(err) {
			return nil, trace.Wrap(err)
		}
```